--- a/glib/gdate.c
+++ b/glib/gdate.c
@@ -56,6 +56,8 @@
 #include "garray.h"
 #endif
 
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+
 /**
  * SECTION:date
  * @title: Date and Time Functions
--- a/gio/gdbusauth.c
+++ b/gio/gdbusauth.c
@@ -44,6 +44,8 @@
 
 #include "glibintl.h"
 
+#pragma GCC diagnostic ignored "-Wformat-overflow"
+
 G_GNUC_PRINTF(1, 2)
 static void
 debug_print (const gchar *message, ...)
--- a/gio/gdbusmessage.c
+++ b/gio/gdbusmessage.c
@@ -54,6 +54,8 @@
 
 #include "glibintl.h"
 
+#pragma GCC diagnostic ignored "-Wformat-overflow"
+
 typedef struct _GMemoryBuffer GMemoryBuffer;
 struct _GMemoryBuffer
 {
